modules/virtualization/microvm/guivm.nix

```diff
@@ -0,0 +1,128 @@
+# Copyright 2022-2023 TII (SSRC) and the Ghaf contributors
+# SPDX-License-Identifier: Apache-2.0
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}: let
+  configHost = config;
+  waypipe-ssh = pkgs.callPackage ../../../user-apps/waypipe-ssh {};
+  guivmBaseConfiguration = {
+    imports = [
+      ({
+        lib,
+        pkgs,
+        ...
+      }: {
+        ghaf = {
+          users.accounts.enable = lib.mkDefault configHost.ghaf.users.accounts.enable;
+          profiles.graphics.enable = true;
+          profiles.applications.enable = false;
+          windows-launcher.enable = false;
+          development = {
+            # NOTE: SSH port also becomes accessible on the network interface
+            #       that has been passed through to NetVM
+            ssh.daemon.enable = lib.mkDefault configHost.ghaf.development.ssh.daemon.enable;
+            debug.tools.enable = lib.mkDefault configHost.ghaf.development.debug.tools.enable;
+          };
+        };
+
+        environment = {
+          etc = {
+            "ssh/waypipe-ssh".source = "${waypipe-ssh}/keys/waypipe-ssh";
+          };
+          systemPackages = [
+            pkgs.waypipe
+          ];
+        };
+
+        networking.hostName = "guivm";
+        system.stateVersion = lib.trivial.release;
+
+        nixpkgs.buildPlatform.system = configHost.nixpkgs.buildPlatform.system;
+        nixpkgs.hostPlatform.system = configHost.nixpkgs.hostPlatform.system;
+
+        networking = {
+          enableIPv6 = false;
+          interfaces.ethint0.useDHCP = false;
+          firewall.allowedTCPPorts = [22];
+          firewall.allowedUDPPorts = [67];
+          useNetworkd = true;
+        };
+
+        microvm = {
+          mem = 2048;
+          hypervisor = "qemu";
+          qemu.bios.enable = false;
+          storeDiskType = "squashfs";
+          interfaces = [
+            {
+              type = "tap";
+              id = "vm-guivm";
```

Comment:
This is minor but should we just DRY this to `id =  "vm-${networking.hostname}"` with this declaration https://github.com/tiiuae/ghaf/pull/243/files#diff-48f3927dbfa0f525b6dd5c7bc57cc323d0b54bdbd60cf71264f84e0714cdd5f5R40

Reply:
Fixed